code/io/key.cpp

```diff
@@ -277,6 +282,37 @@ int key_to_ascii(int keycode )
 		return ascii_table[keycode];
 }
 
+//This is very bad. However, proper handling done with SDL_TextInput events requires the inputbox structure to be changed to work with a different type of event.
+//This will work for basic replacement of chars depending on the Keyboard layout, but not special characters.
+int key_to_ascii_modern(int keycode)
+{
+	bool shifted;
+
+	if (!key_inited) return 255;
+
+	shifted = keycode & KEY_SHIFTED;
+	keycode &= 0xFF;
+
+	const char* keys = SDL_GetKeyName(SDL_GetKeyFromScancode(FS2toSDL[keycode]));
```

Comment:
operator[]

Reply:
This function doesn't exist anymore. It was a first implementation of text input, but was replaced fully by the event in inputbox.cpp, and was thus removed in one of the later commits. But I do agree on the []